### PR TITLE
handle user privileges needed to install application

### PR DESCRIPTION
**Pull request by jala-dx**

### CODE REVIEW

File: cmd/kots/cli/install.go [hunk 1]
```diff
@@ -59,6 +61,44 @@ func InstallCmd() *cobra.Command {
...
+					},
+				}
+
+				resp, err := clientset.AuthorizationV1().SelfSubjectAccessReviews().Create(context.Background(), sar, metav1.CreateOptions{})
+				if err != nil {
+					return errors.Wrap(err, "failed to run subject review")
+				}
+
+				if !resp.Status.Allowed {
+					log.ActionWithoutSpinner("Attempting to grant RBAC permissions not currently held, Please read https://kots.io/vendor/packaging/rbac, try --skip-rbac-check\n")
```
Comment (divolgin) [inline cmd/kots/cli/install.go:96]: This should be just a `log.Error()`

File: cmd/kots/cli/install.go [hunk 2]
```diff
@@ -59,6 +61,44 @@ func InstallCmd() *cobra.Command {
 
 			log := logger.NewCLILogger()
 
+			if !v.GetBool("skip-rbac-check") {
```
Comment (divolgin) [inline cmd/kots/cli/install.go:64]: This check is not needed if ensure-rbac is false: ` && v.GetBool("ensure-rbac")`

**Review by divolgin — CHANGES_REQUESTED** [review]
I think we also need this in `kots admin-console upgrade`

File: cmd/kots/cli/upstream-upgrade.go [hunk 3]
```diff
@@ -113,6 +113,15 @@ func UpstreamUpgradeCmd() *cobra.Command {
 			}
 
 			log := logger.NewCLILogger()
+			if !v.GetBool("skip-rbac-check") {
```
Comment (divolgin) [inline cmd/kots/cli/upstream-upgrade.go:116]: This is not needed in upstream upgrade.  Only in admin-console upgrade

File: cmd/kots/cli/install.go [hunk 4]
```diff
@@ -59,6 +61,15 @@ func InstallCmd() *cobra.Command {
 
 			log := logger.NewCLILogger()
 
+			if !v.GetBool("skip-rbac-check") && v.GetBool("ensure-rbac") {
+				err := CheckRBAC()
+				if err != nil {
+					log.Error(errors.New("Current user has insufficient privileges to install Admin Console. For more information, please visit https://kots.io/vendor/packaging/rbac"))
```
Comment (divolgin) [inline cmd/kots/cli/install.go:67]: Can you add a screenshot of what this looks like in the terminal?

**Review by divolgin — CHANGES_REQUESTED** [review]

File: cmd/kots/cli/install.go (hunk 4 above)
Comment (jala-dx) [reply cmd/kots/cli/install.go:67]: Screen Shot 2021-05-20 at 12 12 42 PM

Comment (jala-dx) [reply cmd/kots/cli/install.go:67]: Screen Shot 2021-05-20 at 12 51 16 PM

Comment (jala-dx) [reply cmd/kots/cli/install.go:67]: Screen Shot 2021-05-20 at 12 53 16 PM

Comment (jala-dx) [reply cmd/kots/cli/install.go:67]: Screen Shot 2021-05-20 at 12 55 29 PM

**Review by divolgin — APPROVED** [review]

**Pull request merged**